File: sort/generateRandomTxt.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main(int argc, char** argv){

    // if (argc != 2){
    //     printf("Please insert text file length");
    //     return 1;
    // }
    
    int n;
    printf("Please insert the number of elements to sort: \n");
    scanf("%d", &n);
    
    // seed
    int seed = time(NULL);

    int i;
    FILE *fp;
    fp = fopen("randomTexts.txt", "w");
    fprintf(fp, "%d\n", n);
    for (i = 0; i < n; i++){
        fprintf(fp, "%c\n", (97 + rand_r(&seed) % 26));
    }
    fclose(fp);
    return 0;
}

File: sort/generateRandomTxt.py
# We are in main implicitly. Don't need to make a main function
import random








n = 0
#print("Please insert the number of elements to sort: \n")
n = int(input("Please insert the number of elements to sort: \n")) # instead of getting a %d, we cast to int

random.seed(None) # use system time as seed



i = 0
f = open("randomTextsPy.txt", "w")
f.write("{numElems}\n".format(numElems = n))
# for i in range(0, n, 1):
for i in range(n):
    f.write("{elem}\n".format(elem = chr(random.randint(97, 97 + 25)))) # 25 because from alphabet 0: a -> alphabet 25: z

f.close()

